test_scripts/Capabilities/PersistingHMICapabilities/032_SDL_request_capability_in_case_is_not_present_in_cache_and_default_files.lua

```diff
@@ -0,0 +1,97 @@
+--  the default capabilities
+
+-- Preconditions:
+-- 1  Value of HMICapabilitiesCacheFile parameter is defined (hmi_capabilities_cache.json) in smartDeviceLink.ini file
+-- 2. HMI capabilities cache file (hmi_capabilities_cache.json) doesn't exist on file system
+-- 3. One of HMI capabilities (VR/TTS/RC/UI etc) is missing in HMICapabilities file (stored in hmi_capabilities.json)
+-- 4. SDL and HMI are started
+-- 5. HMI does not provide one of HMI capabilities (VR/TTS/RC/UI etc)
+-- Sequence
+-- 1. Ignition OFF/ON cycle performed
+--  a. SDL sends appropriate HMI capabilities (VR/TTS/RC/UI/Buttons/VehicleInfo etc) request to HMI
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require('test_scripts/Capabilities/PersistingHMICapabilities/common')
+
+--[[ Local Variables ]]
+local capabilitiesMap = {
+  UI = {
+    GetLanguage = { "language" },
+    GetSupportedLanguages = { "languages" },
+    GetCapabilities = { "displayCapabilities" }
+    },
+  VR = {
+    GetLanguage = { "language" },
+    GetSupportedLanguages = { "languages" },
+    GetCapabilities = { "vrCapabilities" }
+    },
+  TTS = {
+    GetLanguage = { "language" },
+    GetSupportedLanguages = { "languages" },
+    GetCapabilities = { "vrCapabilities" }
+    },
+  Buttons = {
+    GetCapabilities = { "capabilities" }
+  },
+  VehicleInfo = {
+    GetVehicleType = { "vehicleType" }
+  }
+}
+
+--[[ Local Functions ]]
+local function updateHMICaps(pMod, pRequest)
+  local hmiDefaultCap = common.getDefaultHMITable()
+  for key, _ in pairs (hmiDefaultCap) do
+    if key == pMod then
+      hmiDefaultCap[pMod][pRequest] = nil
+      if not pMod == "Buttons" then
+        hmiDefaultCap[pMod].IsReady.params.available = true
+      end
+    end
+  end
+  return hmiDefaultCap
+end
+
+local function updateHMICapabilitiesFile(pMod, pCap)
+  local hmiCapTbl = common.getHMICapabilitiesFromFile()
+  for key, _ in pairs (hmiCapTbl) do
+    if key == pMod then
+      hmiCapTbl[pMod][pCap] = nil
+      if not pMod == "Buttons" then
+        hmiCapTbl[pMod].IsReady.params.available = true
+      end
+    end
+  end
+  common.setHMICapabilitiesToFile(hmiCapTbl)
+end
+
+local function getHMIParams(pMod, pRequest)
+  local hmiCapTbl = common.getHMIParamsWithOutRequests()
+  hmiCapTbl[pMod][pRequest].occurrence = 1
+  return hmiCapTbl
+end
+
+--[[ Scenario ]]
+for mod, capRequests  in pairs(capabilitiesMap) do
+  for req, capabilities  in pairs(capRequests) do
+    for _, cap  in ipairs(capabilities) do
+      common.Title("Preconditions")
+      common.Title("TC processing " .. tostring(mod) .. " " .. tostring(req) .. " " .. tostring(req) .. "]")
+      common.Step("Clean environment", common.preconditions)
+      common.Step("Update HMI capabilities default", updateHMICapabilitiesFile, { mod, cap })
+      common.Step("Ignition on, Start SDL, HMI", common.start, { updateHMICaps(mod, req) })
```

Comment:
@VjKlepikov According proposed changes for `updateHMICaps()` above

```suggestion
      common.Step("Ignition on, Start SDL, HMI", common.start, { updateHMICaps(mod, req, cap) })
```

Reply:
Done (see [15490e2](https://github.com/LuxoftSDL/sdl_atf_test_scripts/pull/29/commits/15490e2165ffad915c784dee5ffdc6665fc428b8) )